tests/tx/test_verification.py

```diff
@@ -0,0 +1,448 @@
+            patch.object(MergeMinedBlock, 'verify_parents', verify_parents_wrapped),
+            patch.object(MergeMinedBlock, 'verify_height', verify_height_wrapped),
+            patch.object(MergeMinedBlock, 'verify_aux_pow', verify_aux_pow_wrapped),
+        ):
+            self.manager.verification_service.verify(block)
+
+        # Block methods
+        verify_pow_wrapped.assert_called_once()
+        verify_no_inputs_wrapped.assert_called_once()
+        verify_outputs_wrapped.assert_called_once()
+        verify_data_wrapped.assert_called_once()
+        verify_sigops_output_wrapped.assert_called_once()
+        verify_parents_wrapped.assert_called_once()
+        verify_height_wrapped.assert_called_once()
+
+        # MergeMinedBlock methods
+        verify_pow_wrapped.assert_called_once()
+
+    def test_transaction_verify_basic(self) -> None:
+        tx = self._get_valid_tx()
+
+        verify_parents_basic_wrapped = Mock(wraps=tx.verify_parents_basic)
+        verify_weight_wrapped = Mock(wraps=tx.verify_weight)
+        verify_pow_wrapped = Mock(wraps=tx.verify_pow)
+        verify_number_of_inputs_wrapped = Mock(wraps=tx.verify_number_of_inputs)
+        verify_outputs_wrapped = Mock(wraps=tx.verify_outputs)
+        verify_sigops_output_wrapped = Mock(wraps=tx.verify_sigops_output)
+
+        with (
+            patch.object(Transaction, 'verify_parents_basic', verify_parents_basic_wrapped),
+            patch.object(Transaction, 'verify_weight', verify_weight_wrapped),
+            patch.object(Transaction, 'verify_pow', verify_pow_wrapped),
+            patch.object(Transaction, 'verify_number_of_inputs', verify_number_of_inputs_wrapped),
+            patch.object(Transaction, 'verify_outputs', verify_outputs_wrapped),
+            patch.object(Transaction, 'verify_sigops_output', verify_sigops_output_wrapped),
+        ):
+            self.manager.verification_service.verify_basic(tx)
+
+        # Transaction methods
+        verify_parents_basic_wrapped.assert_called_once()
+        verify_weight_wrapped.assert_called_once()
+        verify_pow_wrapped.assert_called_once()
+        verify_number_of_inputs_wrapped.assert_called_once()
+        verify_outputs_wrapped.assert_called_once()
+        verify_sigops_output_wrapped.assert_called_once()
+
+    def test_transaction_verify_without_storage(self) -> None:
+        tx = self._get_valid_tx()
+
+        verify_pow_wrapped = Mock(wraps=tx.verify_pow)
+        verify_number_of_inputs_wrapped = Mock(wraps=tx.verify_number_of_inputs)
+        verify_outputs_wrapped = Mock(wraps=tx.verify_outputs)
+        verify_sigops_output_wrapped = Mock(wraps=tx.verify_sigops_output)
+
+        with (
+            patch.object(Transaction, 'verify_pow', verify_pow_wrapped),
+            patch.object(Transaction, 'verify_number_of_inputs', verify_number_of_inputs_wrapped),
+            patch.object(Transaction, 'verify_outputs', verify_outputs_wrapped),
```

Comment:
It's missing a wrapped call to the `verify_number_of_outputs()` which is usually called by `verify_outputs()`.

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/824/commits/00116efab6ba074ef8aad62a3be8198fdfd6d4e5

Reply:
Thanks!